### ISSUE TITLE
CSS Major Overhaul

### ISSUE THREAD

**Issue by nanaya** [ppy/osu-web#53]
As I (not so) recently learned, I've been violating a lot of [efficient css guidelines](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS). I've been trying to reduce the problem by implementing [BEM](http://getbem.com/) in a few places but it's still pretty much over my head (and I think I did it incorrectly).

At least newly written CSS should conform the guidelines and going forward a major rewrite will save a lot of headaches (or so I hope).

Another thing is it's slightly easier doing BEM with LESS (which I just discovered last week):

``` less
.block {
  ...
  &__element { ... } // will turn into .block__element

  &--modifier { ... } // will turn into .block--modifier
}
```

Also, all javascripts related classes should always be prefixed with `js-` so no one (mainly me) won't accidentally break javascripts by moving/renaming/adding things around.

Lastly, visiting the site shouldn't double as system stability test (store products listing page, for example, causes my laptop to go on fire by having it open for a while).

tl;dr: I suck at CSS. Someone pls2halp.

**Comment by claudiodekker** [comment]
After 1.5 hours of discussion and headaches, I think the conclusion on BEM blocks within elements, is that one element should become both the parent block and a child element (mix-in).

For example:
```<ul class="statistics">
        <li class="statistics__item statistic">
                <span class="statistic__item">Ranked Score</span>
                <span class="statistic__item--strong">..</span>
        </li>
        <li class="statistics__item statistic">
                <span class="statistic__item">Hit Accuracy</span>
                <span class="statistic__item--strong">..</span>
        </li>
</ul>```

Please note how the **li**'s here are both statistics__item (elements) and statistic (blocks).
A good explanation of why this is (most probably) the best solution we can come up with can be found  [here] (http://alexbondarev.com/bem-elements-and-their-elements/)

**Comment by kawaii** [comment]
One thing that causes a huge amount of GPU memory load is the moving triangles in the background of the navigation, you could actually see this by removing the node or adblocking it whilst having the show FPS meter enabled (Press escape on the console -> rendering). also
[Google's Optimizing Performance](https://developers.google.com/web/fundamentals/performance/rendering/?hl=en) can be useful to look through.

**Comment by PieInOblivion** [comment]
May I ask why the page is loading a branch of external files from the same domain?
My debug browser keeps bugging me (When I load the page) that 'http://jizz.ppy.sh/forum' (Homepage) is loading an external javascript from a reference and then again in the javascript another CSS file. May I ask why? And I notice this when I load the page, the font changes a couple micro-seconds later as it finished loading the external-external CSS file.

**Comment by nanaya** [comment]
slowly done (and still ongoing) ┐(ﾟ ヮﾟ)┌

The bem seems to be mostly working.
